scripts/create_release_tag.sh

```diff
@@ -112,16 +112,91 @@ function update_pre {
     echo "$major.$minor.$patch-pre-$branch.$commits"
 }
 
+function generate_changelog {
+    prev_tag=$1
+    bazel_target=$2
+
+    bug_changelog=''
+    feature_changelog=''
+    cleanup_changelog=''
+
+    # Find all the commits between now and the last release.
+    commits=$(git log HEAD..."$prev_tag" --pretty=format:"%H")
+
+    # Find all file dependencies of the bazel target.
+    bazel query --noshow_progress 'kind("source file", deps('"$bazel_target"')) union buildfiles(deps('"$bazel_target"'))' | sed  -e 's/:/\//' -e 's/^\/\+//' > target_files.txt
```

Comment:
IMO this expression is brittle and doesn't handle many things such as third party repos etc, might be easier to map from changed files to affected targets like we do in `ci/bazel_build_deps.sh`
Or perhaps easier to just create a set of folders that should be watched based on release kind.

This is a ok starting point but we should revisit this in the future.